### ISSUE TITLE
Pointer Cursor disapears when rdp into 2012R2

### ISSUE THREAD

**Issue by jkuter** [Terminals-Origin/Terminals#2]
After the upgrade to 4.0 my cursor disappears when remote desktop is in use to a server 2012R2 machine and I try to click into a text editor.  To be more clear the cursor is there, I can type, but the pointer is gone and I can select anything in the text window.  The cursor is much smaller but usable outside of the text window like on menu options.  I have changed many of the RDP options to try and get it to work but to no avail.

**Comment by bradthurber** [comment]
Very interesting. I wasn't aware of the existence of "Terminals" until I was having this very same issue with Remote Desktop Connection Manager 2.7 (RDCMan). Thinking it may have something to do with the client I started looking for alternatives and found Terminals. A workaround I was using on RDCMan was to set the Mouse Properties Pointer Scheme to `Windows Standard (large)`. I'm finding this doesn't work for me in Terminals but bumping the scheme up to `Windows Standard (extra large)` does. It would be interesting to learn if this works for you as well.

**Comment by jkuter** [comment]
This changes the cursor size but doesn't have an affect on me being able to type in the text windows.  Its all of terminals for me because I can reproduce it on server 2012r2 and server 2016 and Windows 10.  Some of these are domain machines and others aren't.  Only the 2012 R2 machines have the cursor issue

**Comment by tntvenion** [comment]
I have the same problem that appeared after windows 10 creators update.
The reason for the problem in my case is - I have a 4k monitor. The scaling is set at 125% (probably - I cannot see the exact scaling setting after creators update). Before the update the system (windows) would scale the Terminals application and RDP connections looked good. The cursor is scaled to too small a size after the update (the vertical bar text selector cursor is so small it's invisible). Changing DPI setting for the Terminals application does not help.
The problem is described in this old post (the solution - use RDCman with DPI scaling disabled - did not help me):
https://social.technet.microsoft.com/Forums/en-US/2c4f1b5b-f95b-4e0f-923e-68ff608691ce/small-or-missing-mouse-pointer-in-rdp-session-solved?forum=winserverTS
Now I am left without a custom RDP application that works correctly on my screen (RDCman, mRemoteNG, MobaXterm). Only mstsc manages to scale the cursor correctly.

**Comment by tntvenion** [comment]
I have figured how to fix the problem for my situation. I have 2 monitors: a laptop screen with a 3k resolution and a desktop monitor with a QHD resolution. If I set scaling on the laptop monitor to 150% and desktop - 100% then I will have this problem where the I-beam cursor will be invisible in the RDP session. If I set the laptop monitor to 125% scaling then the I-beam cursor will be visible in RDP. This also fixes some cursor issues on my home desktop word/outlook applications.
This is extremely annoying since I have to user a sub-optimal scaling.

**Comment by mtherien** [comment]
I'm having this problem as well.  If I run the native Remote Desktop Client the problem goes away.  My guess is the problem is in the Terminals app.  I tried changing resolutions without any luck.

**Comment by tntvenion** [comment]
I've had an opportunity to test this problem on a PC with a single 4k monitor. The problem happens only after Win 10 Creators Update and it happens on any scaling setting that is higher than 100%. Different scaling will cause a different effect and sometimes after restarting the PC the problem will go away on some scaling settings (125% in my case). So I restart my PC and check if the problem is there and restart again until I can see the cursor.
The problem affects all of the RDP applications (RDCman, mRemoteNG, MobaXterm, Terminals) other than mstsc. The problem also affects MS Office (word, outlook 2016) applications - the cursor disappears or is cut there as well (the effect sometimes goes away after PC restart as well).
All of the mentioned RDP programs use system scaling. But RDCman is per-monitor-aware when it comes to scaling (just like mstsc). This was the basis for the workaround posted on the technet - if the app implements scaling correctly then there should no be any problems. That that is not true after Win 10 creators update. Or at least the app needs to be updated (RDCman 2.7 was released in 2014, and it was updated at the time to support win 8, 8.1, 2012).
So basically Microsoft did something that broke scaling support for older apps (MS Office as well). But this problem is either an edge case or something else in combination with win 10 CU is causing it since it is not widely reported on technet.

**Comment by JanProchy** [comment]
In Windows 10 and Terminals 4.0.1 solved by setting custom scaling to 100%.

**Comment by dmprutsman** [comment]
I had this same issue with the text selector cursor disappearing while using RDCMan 2.7 on Windows Server 2012R2 and 2016. I resolved it by going into the mouse properties > pointers tab and selecting the scheme named 'Windows Black (system scheme)'.
